tests/test_splitter.py

```diff
@@ -0,0 +1,388 @@
+                   "label2": { "attrs": attr2, "counts": counts }}
+        source = self._generate_dataset(config)
+
+        splits = [('train',.7),('test',.3)]
+        actual = splitter.SplitforClassification(source,splits)
+
+        self.assertEqual(168, len(actual.get_subset('train')))
+        self.assertEqual(72, len(actual.get_subset('test')))
+
+        # check stats for train
+        stat_train = compute_ann_statistics(actual.get_subset('train'))
+        dist_train = stat_train["annotations"]["labels"]["distribution"]
+        self.assertEqual(84, dist_train["label1"][0])
+        self.assertEqual(84, dist_train["label2"][0])
+        attr_train = stat_train["annotations"]["labels"]["attributes"]
+        self.assertEqual(49*2, attr_train["attr1"]["distribution"]["0"][0])
+        self.assertEqual(35*2, attr_train["attr1"]["distribution"]["1"][0])
+        self.assertEqual(28, attr_train["attr2"]["distribution"]["0"][0])
+        self.assertEqual(21, attr_train["attr2"]["distribution"]["1"][0])
+        self.assertEqual(35, attr_train["attr2"]["distribution"]["2"][0])
+        self.assertEqual(28, attr_train["attr3"]["distribution"]["0"][0])
+        self.assertEqual(21, attr_train["attr3"]["distribution"]["1"][0])
+        self.assertEqual(35, attr_train["attr3"]["distribution"]["2"][0])
+
+        # check stats for test
+        stat_test = compute_ann_statistics(actual.get_subset('test'))
+        dist_test = stat_test["annotations"]["labels"]["distribution"]
+        self.assertEqual(36, dist_test["label1"][0])
+        self.assertEqual(36, dist_test["label2"][0])
+        attr_test = stat_test["annotations"]["labels"]["attributes"]
+        self.assertEqual(21*2, attr_test["attr1"]["distribution"]["0"][0])
+        self.assertEqual(15*2, attr_test["attr1"]["distribution"]["1"][0])
+        self.assertEqual(12, attr_test["attr2"]["distribution"]["0"][0])
+        self.assertEqual( 9, attr_test["attr2"]["distribution"]["1"][0])
+        self.assertEqual(15, attr_test["attr2"]["distribution"]["2"][0])
+        self.assertEqual(12, attr_test["attr3"]["distribution"]["0"][0])
+        self.assertEqual( 9, attr_test["attr3"]["distribution"]["1"][0])
+        self.assertEqual(15, attr_test["attr3"]["distribution"]["2"][0])
+
+    def test_split_for_classification_gives_error_on_multi_label(self):
+        iterable = [
+            DatasetItem(
+                id='1',
+                annotations=[Label(0), Label(1)],
+                subset=""
+            ),
+            DatasetItem(
+                id='2',
+                annotations=[Label(0), Label(2)],
+                subset=""
+            )
+        ]
+        categories = {
+            AnnotationType.label: LabelCategories.from_iterable(
+                'label_' + str(label) for label in range(3)
+            )}
+        source = Dataset.from_iterable(iterable, categories)
+        with self.assertRaises(Exception):
```

Comment:
Consider using at least `assertRaisesRegex` to avoid successful test passing in unexpected failures, like `KeyError` etc. .

Reply:
Thank you for letting me know this. I'll change all the occurence.